### PR TITLE
 Aurora jobupdate functionality -- pause/resume/pulse api

### PR DESCRIPTION
**Pull request by kkrishna**
-----------------------------------------
## Please read instructions below ##

Before submitting, please make sure you run a vagrant box running Aurora with the latest version shown in .auroraversion and run go test from the project root.

To run an Aurora Vagrant image, follow the instructions here:
http://aurora.apache.org/documentation/latest/getting-started/vagrant/

* Have you run goformat on the project before submitting? yes

* Have you run go test on the project before submitting? Do all tests pass? yes

* Does the Pull Request require a test to be added to the end to end tests? If so, has it been added? yes, more tests to follow later to cover all apis.

### CODE REVIEW

File: realis.go [hunk 1]
```diff
@@ -859,6 +864,73 @@ func (r *realisClient) AbortJobUpdate(
...
+		})
+
+		if clientErr != nil {
+			return false, clientErr
+		}
+
+		return true, nil
+	})
+	if retryErr != nil {
+		return nil, errors.Wrap(clientErr, retryErr.Error()+": Error sending PauseJobUpdate command to Aurora Scheduler")
```
Comment (rdelval) [inline realis.go]: If clientErr becomes nil here, we'll incorrectly return a nil error. retryErr will become clientErr if clientErr is not nil as per line 877 so we can get away with just wrapping retryErr.

File: realis_e2e_test.go [hunk 2]
```diff
@@ -128,6 +129,91 @@ func TestRealisClient_CreateJob_Thermos(t *testing.T) {
...
+		Disk(100).
+		IsService(true).
+		InstanceCount(1).
+		AddPorts(1).
+		AddLabel("currentTime", time.Now().String())
+
+	pulse := int32(30)
+	timeout := 300
+	settings := realis.NewUpdateSettings()
+	settings.BlockIfNoPulsesAfterMs = &pulse
```
Comment (rdelval) [inline realis_e2e_test.go]: We can use a thrift helper here. `thrift.Int32Ptr(30)`

File: updatejob.go [hunk 3]
```diff
@@ -29,8 +29,8 @@ func NewDefaultUpdateJob(config *aurora.TaskConfig) *UpdateJob {
 
 	req := aurora.NewJobUpdateRequest()
 	req.TaskConfig = config
-	s := NewUpdateSettings().Settings()
-	req.Settings = &s
+	s := NewUpdateSettings()
+	req.Settings = s
```
Comment (rdelval) [inline updatejob.go]: can we do `req.Settings = NewUpdateSettings()` ?

File: realis.go (hunk 1 above)
Comment (kkrishna) [reply realis.go]: @rdelval would there be a case when clientErr is nil while retryErr is not?

Comment (rdelval) [reply realis.go]: Only remote possibility is that we get a TimeoutErr from the exponential backoff without the client getting set to err. I don't know if that code path is even possible, but it's hard to tell from the two closures being used. I had changed this in my PR that's on hold now. I think I will make it a point to simplify our  two closures maybe even combining them into one. As you had mentioned to me offline, this code is confusing.

Comment (kkrishna) [reply realis.go]: sounds good @rdelval .. it is updated.. take care of this.. thanks.

**Pull request merged**